test/entry_test.rb

```diff
@@ -25,82 +25,59 @@
 
   describe ".all" do
     subject { SystemdJournal::Entry.all(args) }
-    let(:cmd) { "LANG=C journalctl --no-pager -o json" }
 
     describe "journalctl invocation" do
-      context "when called with no additional arguments" do
-        let(:args) { nil }
+      context "when called with no options or matches" do
+        let(:args) { {} }
 
         it "invokes journalctl without filters" do
-          expect_to_execute(cmd).and_return(journalctl_result)
+          expect_journalctl_with({"no-pager" => nil, "output" => "json"}, []).
+            and_return(cmd_result_for("journalctl"))
           subject
         end
       end
 
-      context "when called with additional arguments" do
-        let(:args) { "-b" }
+      context "when called with additional options" do
+        let(:args) { { options: { "boot" => 0 } } }
 
         it "passes the arguments to journalctl" do
-          expect_to_execute("#{cmd} -b").and_return(journalctl_result)
+          expect_journalctl_with({"no-pager" => nil, "output" => "json", "boot" => 0}, []).
+            and_return(cmd_result_for("journalctl"))
           subject
         end
       end
-    end
-
-    describe "journalctl parsing" do
-      # Arguments are not relevant, we are going to stub the call
-      let(:args) { nil }
-      # Stub journalctl call
-      before do
-        allow_to_execute(/#{cmd}/).and_return(result)
-      end
 
-      context "when journalctl reports 'Failed to determine timestamp'" do
-        let(:result) {
-          journalctl_error("Failed to determine timestamp: Cannot assign")
-        }
+      context "when called with additional matches" do
+        let(:args) { { matches: ["/dev/sda"] } }
 
-        it "returns an empty array" do
-          expect(subject).to eq([])
+        it "passes the arguments to journalctl" do
+          expect_journalctl_with({"no-pager" => nil, "output" => "json"}, ["/dev/sda"]).
+            and_return(cmd_result_for("journalctl"))
+          subject
         end
       end
+    end
 
-      context "when journalctl reports 'Failed to get realtime timestamp'" do
-        let(:result) {
-          journalctl_error("Failed to get realtime timestamp: Cannot assign")
-        }
+    describe "journalctl parsing" do
+      # Arguments are not relevant, we are going to stub the journalctl call
+      let(:args) { {} }
 
-        it "returns an empty array" do
-          expect(subject).to eq([])
-        end
+      before do
+        allow_to_execute(/journalctl/).and_return(cmd_result_for("journalctl"))
       end
 
-      context "when journalctl reports an unexpected error" do
-        let(:result) {
-          journalctl_error("There are always more ways to crash")
-        }
-
-        it "raises RuntimeError" do
-          expect{subject}.to raise_error(RuntimeError)
-        end
+      it "ignores journalctl markers" do
+        expect(subject.size).to eq(7)
```

Comment:
well, this is quite hard to maintain it because when you need one more line with something tricky that appear in bug, this breaks in strange way. Maybe simple factory like method can help and be more intuitive and easier to maintain, then captured outputs? Also when format changed, it is easier to modify factory to ensure that all tests use new format and do not test against old unused one.